langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/GeminiStreamingResponseBuilder.java

```diff
@@ -0,0 +1,109 @@
+     */
+    public GeminiStreamingResponseBuilder(boolean includeCodeExecutionOutput) {
+        this.includeCodeExecutionOutput = includeCodeExecutionOutput;
+        this.contentBuilder = new StringBuilder();
+        this.functionCalls = new ArrayList<>();
+    }
+
+    /**
+     * Appends a partial response to the builder.
+     *
+     * @param partialResponse the partial response from Gemini AI
+     * @return an Optional containing the text of the partial response, or empty if no valid text
+     */
+    public Optional<String> append(GeminiGenerateContentResponse partialResponse) {
+        if (partialResponse == null) {
+            return Optional.empty();
+        }
+
+        GeminiCandidate firstCandidate = partialResponse.getCandidates().get(0); // TODO: Handle multiple candidates
+
+        updateFinishReason(firstCandidate);
+        updateTokenUsage(partialResponse.getUsageMetadata());
+
+        GeminiContent content = firstCandidate.getContent();
+        if (content == null || content.getParts() == null) {
+            return Optional.empty();
+        }
+
+        AiMessage message = fromGPartsToAiMessage(content.getParts(), this.includeCodeExecutionOutput);
+        updateContentAndFunctionCalls(message);
+
+        return Optional.ofNullable(message.text());
+    }
+
+    /**
+     * Builds the final response from all accumulated partial responses.
+     *
+     * @return a Response object containing the final AiMessage, token usage, and finish reason
+     */
+    public Response<AiMessage> build() {
+        AiMessage aiMessage = createAiMessage();
+        return Response.from(aiMessage, tokenUsage, finishReason);
+    }
+
+    private void updateTokenUsage(GeminiUsageMetadata tokenCounts) {
+        this.tokenUsage = new TokenUsage(
+                tokenCounts.getPromptTokenCount(),
+                tokenCounts.getCandidatesTokenCount(),
+                tokenCounts.getTotalTokenCount()
+        );
+    }
+
+    private void updateFinishReason(GeminiCandidate candidate) {
+        if (candidate.getFinishReason() != null) {
+            this.finishReason = fromGFinishReasonToFinishReason(candidate.getFinishReason());
+        }
+    }
+
+    private void updateContentAndFunctionCalls(AiMessage message) {
+        Optional.ofNullable(message.text()).ifPresent(contentBuilder::append);
+        if (message.hasToolExecutionRequests()) {
+            functionCalls.addAll(message.toolExecutionRequests());
+        }
+    }
+
+    private AiMessage createAiMessage() {
+        String text = contentBuilder.toString();
+        boolean hasText = !text.isEmpty() && !text.isBlank();
+
+        if (hasText && !functionCalls.isEmpty()) {
+            return new AiMessage(text, functionCalls);
+        } else if (hasText) {
+            return new AiMessage(text);
+        } else if (!functionCalls.isEmpty()) {
+            return new AiMessage(functionCalls);
+        }
+        return new AiMessage("No valid return.");
```

Comment:
This does not seem right... When this can happen? Should we throw exception in this case?